src/test/java/edu/ucla/library/iiif/fester/CsvParserTest.java

```diff
@@ -54,7 +55,9 @@ public final void testParse() throws CsvParsingException, CsvException, IOExcept
      */
     @Test
     public final void testParseSpaces() throws CsvParsingException, CsvException, IOException {
-        myCsvParser.parse(getTestPath("spaces.csv"));
+        final CsvMetadata metadata = myCsvParser.parse(getTestPath("spaces.csv")).getCsvMetadata();
+        final String first = metadata.getWorksList().get(8)[0];
```

Comment:
Good, except it's the getWorksList().get(7)[0] record because the first non-header row in that CSV is a "collection" row (so doesn't show up in the `getWorksList()` result). You can confirm by commenting out your `trimValues(row);` code in the `CsvParser` and this test (with `metadata.getWorksList().get(8)[0]`) will still pass.

Reply:
Fixed the off-by-one issue in the test, and made resolution to do better job of learning data structure.